Algorithm.CSharp/CustomDataUsingMapFileRegressionAlgorithm.cs

```diff
@@ -0,0 +1,146 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using QuantConnect.Data;
+using QuantConnect.Data.Custom.SEC;
+using QuantConnect.Data.Market;
+using QuantConnect.Interfaces;
+using QuantConnect.Orders.Fees;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Regression algorithm demonstrating use of map files with custom data
+    /// </summary>
+    /// <meta name="tag" content="using data" />
+    /// <meta name="tag" content="custom data" />
+    /// <meta name="tag" content="regression test" />
+    /// <meta name="tag" content="SEC" />
+    /// <meta name="tag" content="rename event" />
+    /// <meta name="tag" content="map" />
+    /// <meta name="tag" content="mapping" />
+    /// <meta name="tag" content="map files" />
+    public class CustomDataUsingMapFileRegressionAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private Symbol _symbol;
+        private bool _changedSymbol;
+        private bool _properSymbolBeforeRename;
+        private bool _properSymbolAfterRename;
+
+        /// <summary>
+        /// Ticker we use for testing
+        /// </summary>
+        public const string Ticker = "TWX";
+
+        /// <summary>
+        /// Initialise the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.
+        /// </summary>
+        public override void Initialize()
+        {
+            SetStartDate(2001, 1, 1);
+            SetEndDate(2003, 12, 31);
+            SetCash(100000);
+
+            // AOL renames to TWX in 2003
+            _symbol = AddData<SECReport8K>(Ticker, Resolution.Daily).Symbol;
+            AddEquity(Ticker, Resolution.Daily);
+        }
+
+        /// <summary>
+        /// Checks to see if the stock has been renamed, and places an order once the symbol has changed
+        /// </summary>
+        /// <param name="slice"></param>
+        public override void OnData(Slice slice)
+        {
+            if (slice.SymbolChangedEvents.ContainsKey(_symbol))
+            {
+                // Check to see if it was renamed on the 16th
+                _changedSymbol = Time.Date == new DateTime(2003, 10, 16);
+                Log($"{Time} - Ticker changed from: {slice.SymbolChangedEvents[_symbol].OldSymbol} to {slice.SymbolChangedEvents[_symbol].NewSymbol}");
+            }
+
+            foreach (var report in slice.Get<SECReport8K>())
+            {
+                if (!_properSymbolBeforeRename)
```

Comment:
This condition would allow the test to be wrong _and then_ right, and the flag will still be set as passing. Another option is to add these flags to a list and in `OnEndOfAlgorithm` confirm that all of them are correct. Another way is to add just `report.Key.Value` to a list (or dictionary) w/ the dates and perform the checks there.